### PR TITLE
Improve some BN documentation.

### LINKED ISSUE #5066: Error in BN_get_word(3) man page
In the return value documentation:

> BN_get_word() returns the value a, and 0xffffffffL if a cannot be represented as an unsigned long.

Actually, `BN_get_word` returns `BN_MASK2` (not part of public interface) in this case, which is 0xffffffffffffffffL on 64 bit platforms.

See: https://github.com/openssl/openssl/blob/master/doc/man3/BN_zero.pod

### PR DESCRIPTION
**Pull request by richsalz**
Thanks to Nicolas Schodet for pointing this out.

Fixes #5066.  For all releases.

### CODE REVIEW

**Comment by richsalz** [comment]
sorry, one last time please @t-j-h :)

File: doc/man3/BN_zero.pod [hunk 1]
```diff
@@ -14,33 +14,38 @@ operations
...
-BN_get_word() returns B<a>, if it can be represented as an unsigned
-long.
+BN_get_word() returns B<a>, if it can be represented as a B<BN_ULONG>.
 
 =head1 RETURN VALUES
 
-BN_get_word() returns the value B<a>, and 0xffffffffL if B<a> cannot
-be represented as an unsigned long.
+BN_get_word() returns the value B<a>, or all-bits-set if B<a> cannot
+be represented as a single integer.
```
Comment (kroeckx) [inline doc/man3/BN_zero.pod]: So how does one check this in a portable way?

Comment (richsalz) [reply doc/man3/BN_zero.pod]: ```~(BN_LONG)0``` or if retval+1==0

**Comment by kaduk** [comment]
Didn't this whole exercise start out from a complaint that BN_LONG is not a thing in the public headers anymore (and in fact not at all, since commit 94af0cd7f3a8130bbc23feb743b176a74eec7e10, by you)?

**Comment by richsalz** [comment]
I think you read too quickly; BN_ULONG not BN_LONG :)

**Comment by richsalz** [comment]
Merged to master.  Separate PR coming for 1.1.0/1.0.2

**Pull request closed without merging**